Libraries/LibJS/Runtime/Completion.cpp

```diff
@@ -97,13 +97,15 @@ ThrowCompletionOr<Value> await(VM& vm, Value value)
     promise->perform_then(on_fulfilled, on_rejected, {});
 
     // FIXME: Since we don't support context suspension, we attempt to "wait" for the promise to resolve
-    //        by letting the event loop spin until our promise is no longer pending, and then synchronously
-    //        running all queued promise jobs.
-    // Note: This is not used by LibJS itself, and is performed for the embedder (i.e. LibWeb).
+    //        by blocking until all our promise is complete.
     if (auto* custom_data = vm.custom_data()) {
+        // Embedder case (i.e. LibWeb).
         custom_data->spin_event_loop_until(GC::create_function(vm.heap(), [success] {
```

Comment:
spinning the event loop like this technically also runs all promise jobs, but I'm not 100% sure if that's fully correct for all cases

Reply:
The running of queued promise jobs in non-embedder case below is definitely _less_ correct than spinning the event loop since it doesn't do the pop and clear of the execution context stack, but I haven't quite figured out the normative impact of that yet